### PR TITLE
Fix joined cookie name for those containing underline in the suffix

### PR DESCRIPTION
**Pull request by peppered**
## Description
Remove fixed `_0` instead of removing all `_` and `0` chars in the cookie name suffix while joining splitted cookie.

## Motivation and Context
Fixes the https://github.com/oauth2-proxy/oauth2-proxy/issues/960.

## How Has This Been Tested?
The existing tests have been updated to reflect the changes.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by peppered** [comment]
I'm not sure about the checklist. Could someone assist with that?

**Comment by NickMeves** [comment]
> I'm not sure about the checklist. Could someone assist with that?

Looks like you don't need to update any documentation.  Here's the CHANGELOG to add an entry for this PR: https://github.com/oauth2-proxy/oauth2-proxy/blob/master/CHANGELOG.md

File: pkg/sessions/cookie/session_store.go [hunk 1]
```diff
@@ -236,7 +236,7 @@ func joinCookies(cookies []*http.Cookie) (*http.Cookie, error) {
 	for i := 1; i < len(cookies); i++ {
 		c.Value += cookies[i].Value
 	}
-	c.Name = strings.TrimRight(c.Name, "_0")
+	c.Name = strings.TrimSuffix(c.Name, "_0")
```
Comment (NickMeves) [inline pkg/sessions/cookie/session_store.go:239]: So upon a closer look, this function is quite brittle. There's a hidden requirement that the split cookie slice is passed in order.

I think this is cleaner altogether by just adding another parameter to `joinCookies` with the final merged cookie name. The `loadCookie` method that is the only place that calls this already has that as a parameter, so you can just pass it forward.

File: pkg/sessions/cookie/session_store_test.go [hunk 2]
```diff
@@ -147,16 +147,25 @@ func Test_splitCookie_joinCookies(t *testing.T) {
...
-			cookie := &http.Cookie{
-				Name:  strings.Repeat("n", nameSize),
-				Value: value,
-			}
-			splitCookies := splitCookie(cookie)
-			joinedCookie, err := joinCookies(splitCookies)
-			assert.NoError(t, err)
-			assert.Equal(t, *cookie, *joinedCookie)
-		})
+	nameSizes := []int{1, 10, 50, 100, 200, 254}
```
Comment (NickMeves) [inline pkg/sessions/cookie/session_store_test.go:150]: I think we should split these two test scenarios and not try to splice them together.

Makes it cleaner for the size test ensuring nothing breaks at the 256 max & your new trailing suffix test without them impacting eachother potentially.

File: pkg/sessions/cookie/session_store.go (hunk 1 above)
Comment (NickMeves) [reply pkg/sessions/cookie/session_store.go:239]:
```
func joinCookies(cookies []*http.Cookie, cookieName string) (*http.Cookie, error) {
// ...
	c.Name = cookieName
	return c, nil
}
```

Comment (peppered) [reply pkg/sessions/cookie/session_store.go]: Good point. I also added test cases with arbitrary order of the split cookie slice.

**Review by NickMeves — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**